### ISSUE TITLE
Implement Event System

### ISSUE THREAD

**Issue by jokoho48** [nk3nny/LambsDanger#23]
I personally thing an Event system is an important thing, but I am not sure if the CBA Event system is really fitting for this purpose. I would rather have a Mix out of the Vanilla Event System and the CBA one.
a Calling Example would be

FSM:  
This Spawn in this case I needed so that we don't overload the FSM Scheduler time and then we switch back to Unscheduled
```sqf
[_this, group _this, units _this] spawn {
    [{
         ["lambs_danger_OnDanger", _this] call CBA_fnc_localEvent;
    }, _this] call CBA_fnc_directCall;
};
```

UserEnd:
```sqf
["lambs_danger_OnDanger", {
    hint "Awsome Event AI Unit Things " + str _this;
}] call CBA_fnc_addEventHandler;
```

Possible Events:
- OnDanger/OnContact
- OnArtilleryCalled
- OnCQBEntered
- OnPanic
- OnInformationShared
- OnCheckBody

Edited/Update to design we want to use

**Comment by veteran29** [comment]
I also don't see a need to support both event systems at once, this is redundant.

**Comment by nk3nny** [comment]
I suspect anyone savvy enough to leverage eventhandlers will be capable of using the CBA ones. I'm not quite seeing the advantage of the Bohemia one? ELI5

**Comment by jokoho48** [comment]
closed and implemented as of #28
